### PR TITLE
Added examples to the tests, created test for price model

### PR DESCRIPTION
**Pull request by Userlastname**
dev
## JIRA

* [Main JIRA ticket](https://jira.softserve.academy/secure/RapidBoard.jspa?rapidView=id)

## Code reviewers

- [ ] @loqimean 
- [ ] @VasylenchukMischa

## Summary of issue

Create tests 

## Summary of change

Added examples to the tests, created test for price model

## Testing approach

ToDo

## CHECK LIST
- [ ]  СI passed
- [ ]  Сode coverage >=95%
- [ ]  PR is reviewed manually again (to make sure you have 100% ready code)
- [ ]  All reviewers agreed to merge the PR
- [ ]  I've checked new feature as logged in and logged out user if needed
- [ ]  PR meets all conventions

### CODE REVIEW

**Review by mykhailovasylenchuk — APPROVED** [review]

File: app/models/category.rb [hunk 1]
```diff
@@ -2,7 +2,7 @@
 
 class Category < ApplicationRecord
   has_many :category_categoryables, dependent: :restrict_with_exception
-  has_many :categoryables, through: :category_categoryables
+  has_many :categoryables, through: :category_categoryables, source: :category
```
Comment (loqimean) [inline app/models/category.rb:5]: навіщо це?

File: spec/models/calculator_spec.rb [hunk 2]
```diff
@@ -24,6 +24,14 @@
 RSpec.describe Calculator, type: :model do
   subject { build(:calculator) }
 
+  describe "associations" do
+    it { is_expected.to have_attribute(:slug) }
```
Comment (loqimean) [inline spec/models/calculator_spec.rb:28]: безкорисний тест

File: spec/models/calculator_spec.rb [hunk 3]
```diff
@@ -24,6 +24,14 @@
...
+
+    it { is_expected.to have_many(:fields).dependent(:destroy) }
+
+    it { is_expected.to accept_nested_attributes_for(:fields).allow_destroy(true) }
```
Comment (loqimean) [inline spec/models/calculator_spec.rb:32]: впринципі теж безкорисний

File: app/models/category.rb (hunk 1 above)
Comment (Userlastname) [reply app/models/category.rb:5]: Тест не проходив без цього(видавало помилку про відстуність source)

File: spec/models/product_spec.rb [hunk 4]
```diff
@@ -23,6 +23,16 @@
...
   subject { build(:product) }
 
+  describe "associations" do
+    it { is_expected.to belong_to(:product_type) }
+
+    it { is_expected.to have_many(:category_categoryables).dependent(:destroy) }
+
+    it { is_expected.to have_many(:categories).through(:category_categoryables) }
+
+    it { is_expected.to have_many(:prices).dependent(:destroy) }
```
Comment (mykhailovasylenchuk) [inline spec/models/product_spec.rb:33]: по ідеї цих спейсів не треба, можна кожен тест просто в нову лінію

**Review by mykhailovasylenchuk — APPROVED** [review]

File: spec/models/price_spec.rb [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+  describe "associations" do
+    it { is_expected.to belong_to(:priceable) }
+
+    it { is_expected.to belong_to(:category).optional }
+  end
+
+  describe "validations" do
+    it { is_expected.to validate_presence_of(:sum) }
+
+    it { is_expected.to validate_uniqueness_of(:category_id).scoped_to([:priceable_id, :priceable_type]) }
```
Comment (mykhailovasylenchuk) [inline spec/models/price_spec.rb:28]: і тут зроби перенесення

Comment (mykhailovasylenchuk) [inline spec/models/price_spec.rb:28]:
```suggestion
    it { is_expected.to validate_uniqueness_of(:category_id)
      .scoped_to([:priceable_id, :priceable_type]) }
```

**Pull request merged**